module/zfs/dmu_send.c

```diff
@@ -1586,9 +1586,6 @@ send_merge_thread(void *arg)
 		}
 		range_free(front_ranges[i]);
 	}
-	if (range == NULL)
-		range = kmem_zalloc(sizeof (*range), KM_SLEEP);
-	range->eos_marker = B_TRUE;
```

Comment:
I wouldn't be against ASSERT'ing that `range != NULL` and `range->eos_marker, ==, B_TRUE` here.

Reply:
Such assertions will never be triggered. The loop earlier in this function will exit when it sees `range->eos_marker == B_TRUE`. This means both that `range != NULL` and `range->eos_marker == B_TRUE`, so making assertions of them is unnecessary.

Reply:
The motivation for the ASSERT is to clearly document this expected state.  It's not obvious at a glance that `range->eos_marker` must be set at this point.

Reply:
Okay. I will add them and repush.

Reply:
Done.